pkg/allocation/converters/converter.go

```diff
@@ -50,10 +50,19 @@ func ConvertAllocationRequestToGSA(in *pb.AllocationRequest) *allocationv1.GameS
 		}
 	}
 
-	if in.GetMetaPatch() != nil {
+	// Accept both metadata (preferred) and metapatch until metapatch is fully removed.
+	metadata := in.GetMetadata()
+	if metadata == nil {
+		metadata = in.GetMetaPatch()
+	} else if in.GetMetaPatch() != nil {
+		// If both are set, reject the call.
+		return nil
```

Comment:
you need to throw a meaningful exception. The exception should be thrown before making this call at the input validation or you need to return error from the converter.